### PR TITLE
MCA fixes and new triggerSF function

### PR DESCRIPTION
**Pull request by vischia**

### CODE REVIEW

**Comment by vischia** [comment]
Can be merged

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-sigprompt.txt [hunk 1]
```diff
@@ -1,40 +1,42 @@
-ttH+    : TTHnobb_fxfx : xsec : LepGood1_mcMatchId!=0 && LepGood2_mcMatchId!=0 && LepGood3_mcMatchId!=0 && LepGood4_mcMatchId!=0; FillColor=ROOT.kOrange+10
+ttH+    : TTHnobb_fxfx : xsec : LepGood1_isMatchRightCharge && LepGood2_isMatchRightCharge && LepGood3_isMatchRightCharge && LepGood4_isMatchRightCharge ; Label="ttH", FillColor=ROOT.kOrange+10
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-sigprompt.txt]: we should not match by charge in the 4l region (there are no flips contributing)

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-mcfakes.txt [hunk 2]
```diff
@@ -1,31 +1,10 @@
-MC_fakes:  TTJets_DiLepton : xsec : ((LepGood1_mcMatchId==0 && LepGood1_mcPromptGamma==0) || (LepGood2_mcMatchId==0 && LepGood2_mcPromptGamma==0) || (LepGood3_mcMatchId==0 && LepGood3_mcPromptGamma==0) || (LepGood4_mcMatchId==0 && LepGood4_mcPromptGamma==0)) ; FillColor=ROOT.kGray
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-mcfakes.txt:1]: the list of processes should remain the same here... mind that ttZ/WZ, etc may also contribute as fakes to 4l region. In fact, I guess it is more likely than ttbar, which would need two fakes to contribute

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-convs.txt [hunk 3]
```diff
@@ -1,25 +1,22 @@
-Convs  : TTGJets : xsec : ((LepGood1_mcMatchId==0 && LepGood1_mcPromptGamma==1) || (LepGood2_mcMatchId==0 && LepGood2_mcPromptGamma==1) || (LepGood3_mcMatchId==0 && LepGood3_mcPromptGamma==1) || (LepGood4_mcMatchId==0 && LepGood4_mcPromptGamma==1)) ; FillColor=ROOT.kOrange
```
Comment (sscruz) [inline TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-convs.txt:1]: why are TTWToLNu_fxfx, etc removed? please add them back

**Comment by sscruz** [comment]
the changes in the `ttH_plots` look ok to me, some of the mca's need to be fixed, I've added comments

**Comment by sscruz** [comment]
thanks btw (donde hay confianza, da asco :) )

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-sigprompt.txt (hunk 1 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-sigprompt.txt]: Fixed (commit to come)

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-mcfakes.txt (hunk 2 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-mcfakes.txt:1]: Done

File: TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-convs.txt (hunk 3 above)
Comment (vischia) [reply TTHAnalysis/python/plotter/ttH-multilepton/mca-includes/mca-4l-convs.txt:1]: Done, I think

**Comment by sscruz** [comment]
it looks good to me,  thanks!

**Pull request merged**